jerry-core/ecma/base/ecma-gc.c

```diff
@@ -242,9 +242,8 @@ ecma_gc_mark_property (ecma_property_t *property_p) /**< property */
 
       switch (ECMA_PROPERTY_GET_INTERNAL_PROPERTY_TYPE (property_p))
       {
-        case ECMA_INTERNAL_PROPERTY_PRIMITIVE_STRING_VALUE: /* compressed pointer to a ecma_string_t */
-        case ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE: /* compressed pointer to a ecma_number_t */
-        case ECMA_INTERNAL_PROPERTY_PRIMITIVE_BOOLEAN_VALUE: /* a simple boolean value */
+        case ECMA_INTERNAL_PROPERTY_ECMA_VALUE: /* an ecma_value_t except object */
+        case ECMA_INTERNAL_PROPERTY_DATE_FLOAT: /* compressed pointer to a ecma_number_t */
```

Comment:
This is not necessarily a compressed pointer: on 32 bit archs this stores a native ptr.